### PR TITLE
Ldap manual import

### PR DESCRIPTION
**Pull request by MatthieuMan**
acceptance test ok

### CODE REVIEW

File: features/LdapManualImport.feature [hunk 1]
```diff
@@ -0,0 +1,21 @@
+Feature: LDAP_Manual_Import
+    As a company administrator
+    I want to import manually users
+    In order to filter the ones who can access to Centreon Web
+	
+    Background:
+        Given I am logged in a Centreon server
+        And a LDAP configuration has been created 
+        And LDAP authentication is enabled
+        And users auto import is disabled
```
Comment (kduret) [inline features/LdapManualImport.feature]: Too much "And" : 
Given I am logged in a Centreon server
And a LDAP configuration has been created
And users auto import is disabled

File: features/bootstrap/LdapManualImportContext.php [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    public function usersAutoImportIsDisabled()
+    {
+        $this->page = new LdapConfigurationListingPage($this);
+        $this->assertFindLink('openldap')->click();
+        $value = $this->assertFind('css', 'input[name="ldap_auto_import[ldap_auto_import]"]')->getValue();
+
+        if ($value != 0) {
+            throw new Exception('Users auto import enabled');
+        }
+
```
Comment (kduret) [inline features/bootstrap/LdapManualImportContext.php]: be careful about blank lines (coding style)

File: features/bootstrap/LdapManualImportContext.php [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    /**
+     * @Given I search a specific user whose alias contains a special character such as an accent
+     */
+    public function iSearchASpecificUserWhoseAliasContainsASpecialCharacterSuchAsAnAccent()
+    {
+        $this->page = new LdapConfigurationListingPage($this);
+        $this->assertFindLink('openldap')->click();
+        sleep(5);
+        $this->assertFindButton('Import users manually')->click();
+
```
Comment (kduret) [inline features/bootstrap/LdapManualImportContext.php]: be careful about blank lines (coding style)

File: features/bootstrap/LdapManualImportContext.php [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    /**
+     * @Given the LDAP search result displays the expected alias
+     */
+    public function theLdapSearchResultDisplaysTheExpectedAlias()
+    {
+        sleep(5);
+        $this->assertFindButton('Search')->click();
+
```
Comment (kduret) [inline features/bootstrap/LdapManualImportContext.php]: be careful about blank lines (coding style)

File: features/bootstrap/LdapManualImportContext.php [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    /**
+     * @When I import the user
+     */
+    public function iImportTheUser()
+    {
+        sleep(5);
+        $this->assertFind('css', 'input[name="contact_select[select][3]"]')->click();
+        $this->assertFindButton('submitA')->click();
+
```
Comment (kduret) [inline features/bootstrap/LdapManualImportContext.php]: be careful about blank lines (coding style)

File: features/bootstrap/LdapManualImportContext.php [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    /**
+     * @Then the user is created
+     */
+    public function theUserIsCreated()
+    {
+        $this->assertFindLink('centréon-ldap4')->click();
+        $this->page = new ContactConfigurationListingPage($this);
+        $object = $this->page->getEntry($this->alias); 
+        if ($object['alias'] != $this->alias) {
+
```
Comment (kduret) [inline features/bootstrap/LdapManualImportContext.php]: be careful about blank lines (coding style)

File: features/bootstrap/LdapManualImportContext.php [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    public function theUserIsCreated()
+    {
+        $this->assertFindLink('centréon-ldap4')->click();
+        $this->page = new ContactConfigurationListingPage($this);
+        $object = $this->page->getEntry($this->alias); 
+        if ($object['alias'] != $this->alias) {
+            
+            throw new Exception(' contact not created ');
+        }
+
```
Comment (kduret) [inline features/bootstrap/LdapManualImportContext.php]: be careful about blank lines (coding style)

File: features/bootstrap/LdapManualImportContext.php [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+     */
+    public function oneAliasWithAnAccentHasBeenManuallyImported()
+    {
+        $this->aLdapConfigurationHasBeenCreated();
+        $this->ldapAuthenticationIsDisabled();
+        $this->usersAutoImportIsDisabled();
+        $this->iSearchASpecificUserWhoseAliasContainsASpecialCharacterSuchAsAnAccent();
+        $this->theLdapSearchResultDisplaysTheExpectedAlias();
+        $this->iImportTheUser();
+
```
Comment (kduret) [inline features/bootstrap/LdapManualImportContext.php]: be careful about blank lines (coding style)

File: features/bootstrap/LdapManualImportContext.php [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+        $this->iImportTheUser();
+        
+    }
+
+    /**
+     * @When this user logins to Centreon Web
+     */
+    public function thisUserLoginsToCentreonWeb()
+    {
+        $this->iAmLoggedOut();
```
Comment (kduret) [inline features/bootstrap/LdapManualImportContext.php]: I think it's not the better way to check login

File: features/bootstrap/LdapManualImportContext.php [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+    }
+    
+    /**
+     * @Then he's logged by default on Home page
+     */
+    public function hesLoggedByDefaultOnHomePage()
+    {
+        $this->page = new LoginPage($this);
+        $this->page->login($this->alias, 'centreon-ldap4');
+
```
Comment (kduret) [inline features/bootstrap/LdapManualImportContext.php]: be careful about blank lines (coding style)

File: features/bootstrap/LdapManualImportContext.php [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+    /**
+     * @Given LDAP authentication is enabled
+     */
+    public function ldapAuthenticationIsDisabled()
+    {
+        $this->page = new LdapConfigurationListingPage($this);     
+        $object = $this->page->getEntry('openldap');
+
+        if ($object['status'] != 'Enabled') {
+            throw new Exception(' LDAP authentification is disabled');
```
Comment (kduret) [inline features/bootstrap/LdapManualImportContext.php]: fr: authentification
en: authentication

**Comment by kduret** [comment]
Furthermore, use spin method instead of sleep ;)

**Comment by MatthieuMan** [comment]
changes done

**Pull request merged**